waiter/src/waiter/process_request.clj

```diff
@@ -520,6 +535,12 @@
             (confirm-live-connection-without-abort)
             (let [{:keys [service-id service-description] :as descriptor} (request->descriptor-fn request)
                   {:strs [metric-group]} service-description
+                  request (-> request
+                              (assoc :metric-group metric-group
+                                     :service-id service-id
+                                     :service-name (get service-description "name")
+                                     :service-version (get service-description "version"))
```

Comment:
Can we destructure these guys?

Reply:
`name` is a function.  It's actually used inside the current `let` block which is why avoided the destructuring.